jdaviz/app.py

```diff
@@ -1026,6 +1026,13 @@ def add_data_to_viewer(self, viewer_reference, data_label,
             Removes all other currently plotted data and only shows the newly
             defined data set.
         """
+        if 'mosviz_row' in self.state.settings:
+            if not (self.get_viewer("table-viewer").row_selection_in_progress):
+                if (self.data_collection[data_label].meta['mosviz_row'] !=
+                        self.state.settings['mosviz_row']):
```

Comment:
I think all 3 `if`s can be chained together with `and` into just 1 `if`?

Reply:
Duh... Thanks for catching that. I've clearly been staring at these lines for too long and have gotten loopy 😅 Done in https://github.com/spacetelescope/jdaviz/pull/1790/commits/f85fddebf1112f8951816f77797ebdc2de235d89